### PR TITLE
#5354 Port Resnet50 to TTNN

### PR DESCRIPTION
**Pull request by shwetankTT**

### CODE REVIEW

File: tests/ttnn/integration_tests/resnet/resnet50.py [hunk 1]
```diff
@@ -0,0 +1,330 @@
...
+    torch_model.to(torch.bfloat16)
+    torch_input_tensor_test = torch_input_tensor.to(torch.bfloat16)
+    torch_output_tensor = torch_model(torch_input_tensor_test)
+
+    input_tensor = pad_and_fold_conv_activation_for_unity_stride(
+        torch_input_tensor, *torch_model.conv1.padding, *torch_model.conv1.stride
+    )
+    input_tensor = torch.permute(input_tensor, (0, 2, 3, 1))
+    input_tensor = ttnn.from_torch(input_tensor, dtype=ttnn.bfloat16)
+    output_tensor = parameters.conv1.copy_input_to_device(input_tensor)
```
Comment (AleksKnezevic) [inline tests/ttnn/integration_tests/resnet/resnet50.py:253]: this is pretty nitpicky, but maybe you want to move the model implementation to `ttnn_functional_resnet.py`?

Comment (shwetankTT) [reply tests/ttnn/integration_tests/resnet/resnet50.py:253]: Noted.

**Review by arakhmati — CHANGES_REQUESTED** [review]
I added the call to downsample directly inside of ttnn.Conv2d, so you don't need to expose this op

**Comment by shwetankTT** [comment]
> I added the call to downsample directly inside of ttnn.Conv2d, so you don't need to expose this op

Thanks I will ping you for more info.

**Comment by mywoodstock** [comment]
@shwetankTT nit-picking: can you replace all `Bottol` with `Bottle` :)

**Comment by eyonland** [comment]
Would it make sense to duplicate this functional version and go ahead and add them both to [https://github.com/tenstorrent-metal/tt-metal/blob/main/tests/scripts/run_performance.sh](https://github.com/tenstorrent-metal/tt-metal/blob/main/tests/scripts/run_performance.sh)?   The idea is that as progress as made we would all see the performance improving daily.

**Review by arakhmati — APPROVED** [review]

File: tests/ttnn/integration_tests/resnet/test_ttnn_functional_resnet.py [hunk 2]
```diff
@@ -277,3 +299,223 @@ def custom_preprocessor(model, name, ttnn_module_args, convert_to_ttnn):
 
     # The check below doesn't work yet
     # assert_with_pcc(torch_output_tensor, output_tensor, pcc=0.999)
+
+
+@skip_for_wormhole_b0()
+def test_bottolneck_block(device):
```
Comment (eyonland) [inline tests/ttnn/integration_tests/resnet/test_ttnn_functional_resnet.py:305]: would you fix the spelling here?

Comment (eyonland) [reply tests/ttnn/integration_tests/resnet/test_ttnn_functional_resnet.py:305]: test_bottleneck_block(device)

**Review by eyonland — APPROVED** [review]
Please just fix the spelling on the test!

**Review by arakhmati — APPROVED** [review]

**Review by arakhmati — APPROVED** [review]

**Comment by shwetankTT** [comment]
Already merged with mainline.

**Pull request closed without merging**